### ISSUE TITLE
Version inconsistency in python interface between get_version() and __version__

### ISSUE THREAD

**Issue by atztogo** [spglib/spglib#478]
I used a spglib develop branch (69da2b8dc41e) installed via `pip install -e .`
```
In [1]: import spglib

In [2]: spglib.get_version()
Out[2]: (2, 3, 1)

In [3]: spglib.__version__
Out[3]: '2.3.2.dev93+g69da2b8'
```
Is this expected behavor?

I got `ValueError` in seekpath at

https://github.com/giovannipizzi/seekpath/blob/1e590ceacb4ba36e59a87ed2e8ca95af15d6b567/seekpath/hpkot/tools.py#L234

I encountered inconvenience when using seekpath. Seekpath is a popular package to suggest a reasonable Brillouin zone special path for each space group type.

If this is an expected behavior, I will make a PR to seekpath.

**Comment by LecrisUT** [comment]
Yes that is expected. Setuptools-scm bumps the patch version when you are away from a git tag [`guess-next-dev`](https://setuptools-scm.readthedocs.io/en/latest/extending/#available-implementations). What is a bit weird is that the C library version seems to be a bit mismatched. Can you check that `spglib.spg_get_version_full()` is equivalent?

As for Giovanni's code, it is recommended to move to `imortlib.metadata` + `packaging.version` (applicable to all libraries) or `spglib.__version_tuple__` if the python version is to be compared, or `spglib.spg_get_version_...` if the spglib C library is more relevant.

**Comment by atztogo** [comment]
Thanks @LecrisUT. This is the result. I found my mistake. What I explained was on my fork (`atztogo/spglib`), and I didn't know I had to sync tags for fork. Now it is like below, and it probably works as you expected.

```
In [1]: import spglib

In [2]: spglib.spg_get_version()
Out[2]: '2.4.1'

In [3]: spglib.spg_get_version_full()
Out[3]: '2.4.1.dev3+g69da2b8'

In [4]: spglib.__version__
Out[4]: '2.4.1.dev3+g69da2b8'

In [5]: spglib.__version_tuple__
Out[5]: (2, 4, 1, 'dev3', 'g69da2b8')
```

For seekpath, it only checks if it is not using very old version of spglib, so strict check is unnecessary.

**Comment by LecrisUT** [comment]
So how relevant is this anymore now? For seekpath, here is a snippet to use that is more robust (works with any spglib version):
```python
import sys
if sys.version_info < (3, 8):
    from importlib_metadata import version
else:
    from importlib.metadata import version
from packaging.version import Version

spg_version = Version(version("spglib"))
if spg_version.major > 2:
    pass
```

**Comment by atztogo** [comment]
Thanks. I have forgotten about this. I will make PR.
